### ISSUE TITLE
Change font?

### ISSUE THREAD

**Issue by makslevental** [pavankataria/SwiftDataTables#39]
Is there a way to change the fonts without messing with the xibs?

**Comment by altagir** [comment]
by using apperances

**Comment by makslevental** [comment]
@altagir what does this mean?

**Comment by pavankataria** [comment]
Yeah so you can either use either the UIAppearance class as altagir mentioned though I do think there's a strong case to allow the user to provide their own custom view and this could solve so many appearance related requests. 

I just don't know the best way to deal with resizing since the data table currently manually resizes each row. I just haven't thought about how this would work in relation to laying out the custom views rows and populating the row itself since it's a custom one.. 

It's a case of deciding who is going to be responsible for populating the row. You could argue if you provide a custom view then the library won't pass in the data itself and that it's up to the user to do that. 

Any thoughts?

**Comment by madhavajay** [comment]
It's a real pity, I was trying to use this but unfortunately the inability to change fonts and font sizes and control if the font text wraps or gets shrunk made this unusable. So plus +1 to having more control over the text in each cell.

**Comment by pavankataria** [comment]
Was hoping this would generate some noise but it's been relatively quiet. I'm hoping there's a clean way to allow text wrapping of cells in SwiftDataTable. This does require working out the height of each cell. If this is possible using the Autolayout engine to determine the size that would be best.

**Comment by altagir** [comment]
custom cell with automatic dimension `tableview.rowheight = UITableViewAutomaticDimensionn`

then use autolayout in a custom UITableViewCell to constraint height

ideally SwiftDataTable should provide a setter to set a custom nib

**Comment by pavankataria** [comment]
So I'm not using tableviews here, and SwiftDataTables uses collection views with fixed heights defined. Using auto layout engine to determine the sizes of each row is the ideal scenario. 
Though while scrolling, as you don't want to calculate sizes of all the rows since they're not all presented. 

I think the first step is to allow custom cells, though the issue still remains, these custom views will be squashed to the cell attributes defined by `SwiftDataTables`.  What people want are word wrapped cells that automatically resize and take the necessary height, and to apply that height to the reset of the cells in that row. 

If anyone wants to take a stab at implementing autolayout self resizing cells that would be good.

**Comment by PerplexedFox** [comment]
Hi, @pavankataria  so I would like to be able to change the font size of the cells and headers, but keeping other SwiftDataTable-unrelated labels as they were. What is the best way to do it?

**Comment by pavankataria** [comment]
Sorry for the delay. In the upcoming release you can say hello to Autolayout custom self-sizing cells! 
If you go to the demo in the example app I created: "Custom Cells + Auto Height", you'll be able to see it in action with an interactive demo!

<img width=40% height=40% alt="Image" src="https://github.com/user-attachments/assets/4a150e01-8742-4a7a-bd0d-d4c36601b84c" />

This will be available in an upcoming release.
